lib/routes/charge/sale_view.dart

```diff
@@ -0,0 +1,212 @@
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/pos_catalog/actions.dart';
+import 'package:breez/bloc/pos_catalog/bloc.dart';
+import 'package:breez/bloc/pos_catalog/model.dart';
+import 'package:breez/routes/charge/currency_wrapper.dart';
+import 'package:breez/widgets/loader.dart';
+import 'package:breez/widgets/single_button_bottom_bar.dart';
+import 'package:flutter/material.dart';
+import 'package:breez/widgets/back_button.dart' as backBtn;
+
+import 'items/item_avatar.dart';
+
+class SaleView extends StatefulWidget {
+  final bool useFiat;
+
+  const SaleView({Key key, this.useFiat}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() {
+    return SaleViewState();
+  }
+}
+
+class SaleViewState extends State<SaleView> {
+
+  StreamSubscription<Sale> _currentSaleSubscrription;
+
+  @override void didChangeDependencies() {
+    if (_currentSaleSubscrription == null) {
+      PosCatalogBloc posCatalogBloc =
+        AppBlocsProvider.of<PosCatalogBloc>(context);
+      _currentSaleSubscrription = posCatalogBloc.currentSaleStream.listen((sale) {
+        if (sale.saleLines.length == 0) {
+          Navigator.of(context).pop();
+        }
+      });
+    }
+    super.didChangeDependencies();
+  }
+
+  @override 
+  void dispose() {
+    _currentSaleSubscrription.cancel();
+    super.dispose();
+  }
+  
+  @override
+  Widget build(BuildContext context) {
+    PosCatalogBloc posCatalogBloc =
+        AppBlocsProvider.of<PosCatalogBloc>(context);
+    AccountBloc accountBloc = AppBlocsProvider.of<AccountBloc>(context);
+
+    return StreamBuilder<Sale>(
+        stream: posCatalogBloc.currentSaleStream,
+        builder: (context, saleSnapshot) {
+          var currentSale = saleSnapshot.data;
+          return StreamBuilder<AccountModel>(
+              stream: accountBloc.accountStream,
+              builder: (context, accSnapshot) {
+                var accModel = accSnapshot.data;
+                if (accModel == null) {
+                  return Loader();
+                }
+                return Scaffold(
+                  appBar: AppBar(
+                    iconTheme: Theme.of(context).appBarTheme.iconTheme,
+                    textTheme: Theme.of(context).appBarTheme.textTheme,
+                    backgroundColor: Theme.of(context).canvasColor,
+                    leading: backBtn.BackButton(iconData: Icons.close),
+                    title: _TotalSaleCharge(
+                      accountModel: accModel,
+                      currentSale: currentSale,
+                      useFiat: widget.useFiat,
+                    ),
+                    actions: <Widget>[
+                      IconButton(
+                        icon: Icon(
+                          Icons.note_add,
+                          color: Theme.of(context).iconTheme.color,
+                        ),
+                        onPressed: () {},
```

Comment:
Comment out note button till Note PR, unless this was added for testing purposes.

Reply:
Let's leave it so I can get feedback on the UI.